server/command/service.go

```diff
@@ -30,52 +30,139 @@ type service struct {
+		},
+		"list": {
+			f:            s.executeList,
+			autoComplete: model.NewAutocompleteData("list", "", "List installed and registered apps"),
+		},
+		"uninstall": {
+			f:            s.checkSystemAdmin(s.executeUninstall),
+			autoComplete: uninstallAC,
+		},
 	}
 
-	sort.Strings(subTrigger)
+	conf := s.conf.GetConfig()
 
-	helpText := "Available commands: "
-	for i, t := range subTrigger {
-		if i == 0 {
-			helpText += t
-		} else {
-			helpText += ", " + t
+	if conf.DeveloperMode {
+		debugAddManifestAC := model.NewAutocompleteData("debug-add-manifest", "", "Add a manifest to the local list of known apps")
+		debugAddManifestAC.AddNamedTextArgument("url", "URL of the manifest to add", "URL", "", true)
+
+		all["debug-bindings"] = commandHandler{
+			f:            s.executeDebugBindings,
+			devOnly:      true,
+			autoComplete: model.NewAutocompleteData("debug-bindings", "", "List bindings"),
+		}
+		all["debug-clean"] = commandHandler{
+			f:            s.executeDebugClean,
+			devOnly:      true,
+			autoComplete: model.NewAutocompleteData("debug-clean", "", "Delete all KV data"),
+		}
+		// TODO ticket: change to watch-manifest
+		all["debug-add-manifest"] = commandHandler{
+			f:            s.executeDebugAddManifest,
+			devOnly:      true,
+			autoComplete: debugAddManifestAC,
 		}
 	}
 
-	autoComplete := model.NewAutocompleteData(config.CommandTrigger, "[command]", helpText)
+	all["install"] = s.installCommand(conf)
 
-	for _, t := range subTrigger {
-		c := subCommands[t]
-		if c.debug && !conf.DeveloperMode {
-			continue
-		}
+	return all
+}
+
+func (s *service) installCommand(conf config.Config) commandHandler {
+	h := commandHandler{
+		f: s.checkSystemAdmin(s.executeInstall),
+		autoComplete: &model.AutocompleteData{
+			Trigger:  "install",
+			HelpText: "Install an App.",
+			RoleID:   model.SYSTEM_ADMIN_ROLE_ID,
+			Arguments: []*model.AutocompleteArg{
+				{
+					Name:     "app-secret",
+					HelpText: "(HTTP) App's JWT secret used to authenticate incoming messages from Mattermost.",
+					Type:     model.AutocompleteArgTypeText,
+					Data: &model.AutocompleteTextArg{
+						Hint: "secret string",
+					},
+				},
+			},
+		},
+	}
 
-		autoComplete.AddCommand(c.autoComplete)
+	if conf.MattermostCloudMode {
+		// install only by ID (from the marketplace) in cloud mode
+		h.autoComplete.Arguments = append(h.autoComplete.Arguments, &model.AutocompleteArg{
+			HelpText: "ID of the app to install",
+			Type:     model.AutocompleteArgTypeText,
+			Data: &model.AutocompleteTextArg{
+				Hint: "secret string",
+			},
+			Required: true,
+		})
+	} else {
+		// install from URL in the on-prem mode
+		h.autoComplete.Arguments = append(h.autoComplete.Arguments, &model.AutocompleteArg{
+			Name:     "url",
+			HelpText: "URL of the App's manifest",
+			Type:     model.AutocompleteArgTypeText,
+			Data: &model.AutocompleteTextArg{
+				Hint: "URL",
+			},
+			Required: true,
```

Comment:
Why is an URL requires for on-prem?

Reply:
No longer relevant - it's now the only way in on-prem